src/Umbraco.Core/Composing/DisableApplicationComponentAttribute.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+
+namespace Umbraco.Core.Composing
+{
+    /// <summary>
+    /// Indicates that an <see cref="IApplicationComponent" /> should be disabled.
+    /// </summary>
+    /// <remarks>
+    /// This attribute has greater priority than the <see cref="EnableAttribute" /> when it's marking the component itself, but lower priority when it's referencing another.
+    /// </remarks>
+    /// <seealso cref="System.Attribute" />
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    public class DisableApplicationComponentAttribute : Attribute
```

Comment:
Just like the `DisableComposerAttribute`, this can be added as an assembly attribute to disable the specified `IApplicationCompontent` type.